### ISSUE TITLE
Firefox: the focus-visible version of `:focus-within` fails

### ISSUE THREAD

**Issue by masakudamatsu** [masakudamatsu/masakudamatsu.dev#12]
Firefox fails to show any focus ring on pills with link.

**Comment by masakudamatsu** [comment]
## Problem

Firefox is yet to support `:has()` selector ([Can I Use?](https://caniuse.com/css-has) as of Sep 2023. Therefore, the `:has(:focus-visible)` fails to work.

**Comment by masakudamatsu** [comment]
## Solution

Make the focus-visible version of `:focus-within` a progressive enhancement:
![carbon-6](https://github.com/masakudamatsu/masakudamatsu.dev/assets/36348559/089025a3-b9eb-4fbf-9082-92044385116d)
